util/collate/collate.go

```diff
@@ -134,31 +134,28 @@ func CollationID2Name(id int32) string {
 type binPaddingCollator struct {
 }
 
-func (bpc *binPaddingCollator) Compare(a, b string, opt CollatorOption) int {
-	aLen := len(a)
-	bLen := len(b)
-	noPaddingResult := 0
-	if aLen > bLen {
-		noPaddingResult = strings.Compare(a[:bLen], b)
-		if noPaddingResult != 0 {
-			return noPaddingResult
-		}
-		return strings.Compare(a[bLen:], strings.Repeat(" ", aLen-bLen))
-	} else if aLen < bLen {
-		noPaddingResult = strings.Compare(a, b[:aLen])
-		if noPaddingResult != 0 {
-			return noPaddingResult
+func truncateTailingSpaceTwo(a, b string) (string, string) {
```

Comment:
I don't think we need `truncateTailingSpaceTwo`. Just replace it with two `truncateTailingSpaceOne` is enough.

Reply:
Done